src/SFA.DAS.IdentifyDataLocks.Domain/CollectionPeriodMappingExtensions.cs

```diff
@@ -66,14 +61,21 @@ public static class CollectionPeriodMappingExtensions
                 PriceStart = earning.PriceEpisodes.FirstOrDefault()?.StartDate,
                 StoppedOn = earning.PriceEpisodes.FirstOrDefault()?.ActualEndDate,
                 IlrSubmissionDate = earning.IlrSubmissionDateTime,
-                Tnp1 = GetTnpValue(earning.PriceEpisodes.FirstOrDefault()?.TotalNegotiatedPrice1),
-                Tnp2 = GetTnpValue(earning.PriceEpisodes.FirstOrDefault()?.TotalNegotiatedPrice2),
-                Tnp3 = GetTnpValue(earning.PriceEpisodes.FirstOrDefault()?.TotalNegotiatedPrice3),
-                Tnp4 = GetTnpValue(earning.PriceEpisodes.FirstOrDefault()?.TotalNegotiatedPrice4)
+                Tnp1 = GetTnpValue(earning, x => x.TotalNegotiatedPrice1),
+                Tnp2 = GetTnpValue(earning, x => x.TotalNegotiatedPrice2),
+                Tnp3 = GetTnpValue(earning, x => x.TotalNegotiatedPrice3),
+                Tnp4 = GetTnpValue(earning, x => x.TotalNegotiatedPrice4),
             };
 
-        private static string GetTnpValue(decimal? tnp) => 
-            tnp?.ToString("c") ?? "-";
+        private static List<AmountFromDate> GetTnpValue(
+            EarningEventModel earning,
+            Func<EarningEventPriceEpisodeModel, decimal> tnpSelector)
+        {
+            return earning.PriceEpisodes.Select(AmountSelector).ToList();
+
+            AmountFromDate AmountSelector(EarningEventPriceEpisodeModel pe) =>
```

Comment:
This threw me off a bit . Can we have the in-line function defined before it is invoked.

Reply:
In C# we generally don't require class-level methods to be declared above their points of use, I see this as the same.

My preference is to see use of functions before their definition, with the rationale that I normally care most about what's going on and less about how it's being achieved.  Until I do care of course, and then I go looking for the function definition.